In [ ]:
import pandas as pd
from statsmodels.tsa.stattools import adfuller, kpss

# Load the dataset
data = pd.read_csv('chart.csv')

# Convert 'DateTime' to datetime format
data['DateTime'] = pd.to_datetime(data['DateTime'])

# Clean the data for 'NIFTY 50'
data_clean = data[['DateTime', 'NIFTY 50']].dropna()
data_clean.set_index('DateTime', inplace=True)

# Clean the data for 'Pre Open NIFTY 50'
data_pre_open = data[['DateTime', 'Pre Open NIFTY 50']].dropna()
data_pre_open.set_index('DateTime', inplace=True)

# Function to perform KPSS test
def kpss_test(series, regression='c'):
    from statsmodels.tsa.stattools import kpss
    kpss_stat, p_value, n_lags, critical_values = kpss(series, regression=regression)
    return {
        "KPSS Statistic": kpss_stat,
        "p-value": p_value,
        "Critical Values": critical_values,
        "Number of Lags": n_lags
    }

# ADF Test Results
def adf_test(series):
    adf_result = adfuller(series)
    return {
        "ADF Statistic": adf_result[0],
        "p-value": adf_result[1],
        "Critical Values": adf_result[4]
    }

# Perform ADF and KPSS Tests

print("NIFTY 50:")
print("ADF Test:", adf_test(data_clean['NIFTY 50']))
print("KPSS Test:", kpss_test(data_clean['NIFTY 50']))

print("\nPre Open NIFTY 50:")
print("ADF Test:", adf_test(data_pre_open['Pre Open NIFTY 50']))
print("KPSS Test:", kpss_test(data_pre_open['Pre Open NIFTY 50']))


NIFTY 50:
ADF Test: {'ADF Statistic': -0.11323264691771068, 'p-value': 0.9481044560885789, 'Critical Values': {'1%': -3.4307876621798616, '5%': -2.8617334276969824, '10%': -2.566872956903668}}
KPSS Test: {'KPSS Statistic': 16.921675939799037, 'p-value': 0.01, 'Critical Values': {'10%': 0.347, '5%': 0.463, '2.5%': 0.574, '1%': 0.739}, 'Number of Lags': 73}

Pre Open NIFTY 50:
ADF Test: {'ADF Statistic': -4.976386534164731, 'p-value': 2.4709423847092437e-05, 'Critical Values': {'1%': -3.4450642060791097, '5%': -2.868027634065253, '10%': -2.5702256691640515}}
KPSS Test: {'KPSS Statistic': 2.251793428535948, 'p-value': 0.01, 'Critical Values': {'10%': 0.347, '5%': 0.463, '2.5%': 0.574, '1%': 0.739}, 'Number of Lags': 12}


C:\Temp\ipykernel_4268\4129356822.py:21: InterpolationWarning: The test statistic is outside of the range of p-values available in the
look-up table. The actual p-value is smaller than the p-value returned.

  kpss_stat, p_value, n_lags, critical_values = kpss(series, regression=regression)
C:\Temp\ipykernel_4268\4129356822.py:21: InterpolationWarning: The test statistic is outside of the range of p-values available in the
look-up table. The actual p-value is smaller than the p-value returned.

  kpss_stat, p_value, n_lags, critical_values = kpss(series, regression=regression)


ADF Test:

Null Hypothesis (H₀): The series is non-stationary.
If p-value < 0.05, reject H₀ → the series is stationary.
KPSS Test:

Null Hypothesis (H₀): The series is stationary.
If p-value < 0.05, reject H₀ → the series is non-stationary.

Combined Analysis:

If ADF indicates stationarity (rejects non-stationarity) and KPSS fails to reject stationarity, the series is stationary.
If ADF indicates non-stationarity (does not reject H₀) or KPSS rejects stationarity, the series is non-stationary.

NIFTY 50:
ADF Test:
p-value = 0.948 → Fail to reject H₀ → Non-Stationary.
KPSS Test: 
If p-value < 0.05 → Reject H₀ → Non-Stationary.

NIFTY 50 is non-stationary. 